### ISSUE TITLE
`enabled` is ignored on azuread_application.app_role

### ISSUE THREAD

**Issue by dimbleby** [hashicorp/terraform-provider-azuread#470]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureAD Provider) Version

terraform 1.0.1, azuread 1.6.0

### Affected Resource(s)

* `azuread_application`

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    azuread = {
      source  = "hashicorp/azuread"
      version = "1.6.0"
    }
  }
}

provider "azuread" {}

resource "azuread_application" "example" {
  display_name = "example-app"

  app_role {
    enabled              = false
    allowed_member_types = ["User"]
    description          = "role description"
    display_name         = "example-role"
    value                = "example"
  }
}
```

### Expected Behavior

The app role created by this configuration should not be enabled.

### Actual Behavior

The app role created by this configuration is enabled.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Replacing `enabled` with `is_enabled` gives a disabled role as expected.  But it also produces a warning telling me that `is_enabled` is deprecated.  (Which is why I tried switching to `enabled`):

```
╷
│ Warning: Deprecated Attribute
│
│   with azuread_application.example,
│   on main.tf line 12, in resource "azuread_application" "example":
│   12: resource "azuread_application" "example" {
│
│ [NOTE] This attribute has been renamed to `enabled` and will be removed in version 2.0 of the AzureAD provider
│
│ (and one more similar warning elsewhere)
╵
```

**Comment by manicminer** [comment]
Hi @dimbleby, many thanks for reporting this bug. Whilst I am able to reproduce the same behavior, we will unfortunately be unable to fix it until version 2.0. This is due to limitations of the Terraform Plugin SDK which requires a breaking change to the schema for applications in order to work around.

This has however, been resolved for v2.0 and you will no longer see this behavior with the `enabled` property. In the meantime, I'd recommend continuing to use the `is_enabled` property, or consider omitting/removing app roles rather than disabling them. The deprecation warning is merely informational and does not otherwise affect your configuration.

**Comment by dimbleby** [comment]
Fair enough.  Should you remove the deprecation warning meanwhile?  I'm not sure about this:

- on the one hand, the warning is just a trap for people who pay attention to such things
- on the other, you really do want to remove the field...

I think on the whole I'd favour removing the warning.
